packages/webamp-modern-2/src/maki/interpreter.ts

```diff
@@ -608,9 +621,11 @@ class Interpreter {
               throw new Error("Tried to add non-numbers.");
           }
           switch (b.type) {
+            case "BOOLEAN": //x2nie
+              b.value = b.value ? 1 : 0;
```

Comment:
This seems weird. `&&` should not modify a value on the stack.

Reply:
done. I revert it back. wa is still working.

Reply:
You have rejected it (and I have removed the patch).
Now, how we can understand this contradiction? :
Logical AND should work with Number and Bool, isn't it?
but here you give a disaster trash error message: "add number" ! what add number  ?
There is tons of same useless message of error. Is it fine?
```
// logAnd (&&)
        case 80: {
          const a = this.stack.pop();
          const b = this.stack.pop();
          // Some of these are probably valid, but we'll enable them once we see usage.
          switch (a.type) {
            case "STRING":
            case "OBJECT":
            case "BOOLEAN":
            case "NULL":
              throw new Error("Tried to add non-numbers.");
          }
          switch (b.type) {
            case "STRING":
            case "OBJECT":
            case "BOOLEAN":
            case "NULL":
              throw new Error("Tried to add non-numbers.");
          }
          if (b.value && a.value) {
            this.push(a);
          } else {
            this.push(b);
          }
          break;
        }       
```

Reply:
press the play button, and endless error begun